be/src/exec/pipeline/pipeline_driver.h

```diff
@@ -342,6 +342,9 @@ class PipelineDriver {
         if (sink_operator()->is_finished()) {
             return true;
         }
+        if (source_operator()->is_epoch_finished() || sink_operator()->is_epoch_finished()) {
```

Comment:
Why change this, since stream operators are driven by the `StreamPipelineDriver` ?

Reply:
Because `is_not_blocked ` is used by PipelineDriverPoller, because `EPOCH_PENDING_FINISHED` or `INPUT_EMPTY` state, one driver may be put into blocked queue.

when one driver is in the blocked queue, need to check `is_not_blocked ` method to check whether the driver is not blocked.

Do you have a better advise to better change this?

Reply:
I see. The problem is these two kinds of operators share the same poller. You could leave a TODO here to improve it in the future.